docs/source/gs-processing/usage/emr.rst

```diff
@@ -77,177 +77,93 @@ inline policy:
-        --region ${REGION} \
-        --release-label ${EMR_VERSION} \
-        --use-default-roles
-
-Running the above will return a JSON structure like:
-
-.. code-block:: json
-
-    {
-        "ClusterId": "j-XXXXXXXXXX",
-        "ClusterArn": "arn:aws:elasticmapreduce:us-east-1:<ACCOUNT>:cluster/j-XXXXXXXXXX"
-    }
-
-Make note of the cluster ID, which we will use to log into the leader instance.
-
-We can also run a waiter to ensure we only proceed to the next step when the cluster is
-ready to run jobs:
-
-.. code-block:: bash
-
-    aws emr wait cluster-running --cluster-id j-XXXXXXXXXX --region ${REGION} && echo "Cluster ready"
-
-Log in to the leader and submit a GSProcessing job
---------------------------------------------------
-
-To submit a job we can use a helper ``bash`` script, which we list below:
-
-.. code-block:: bash
-
-    # submit-gsp-job.sh
-    #!/usr/bin/env bash
-    set -euox pipefail
 
+    # GSProcessing arguments
     MY_BUCKET="enter-your-bucket-name-here"
     REGION="bucket-region" # e.g. us-west-2
     INPUT_PREFIX="s3://${MY_BUCKET}/gsprocessing-input"
-    NUM_EXECUTORS=2
     OUTPUT_BUCKET=${MY_BUCKET}
     GRAPH_NAME="small-graph"
     CONFIG_FILE="gconstruct-config.json"
+    DO_REPARTITION="true"
+    GENERATE_REVERSE="true"
 
-    ACCOUNT=$(aws sts get-caller-identity --query Account --output text)
-
-    REPOSITORY="graphstorm-processing-emr"
-    ARCH="x86_64"
-    TAG="latest-${ARCH}"
-    IMAGE="${ACCOUNT}.dkr.ecr.${REGION}.amazonaws.com/${REPOSITORY}:${TAG}"
 
+    # We assume this script is saved in the same path as submit_gsp_emr_step.py
+    SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
+    # Upload the entry point to S3
+    PATH_TO_ENTRYPOINT="$SCRIPT_DIR/../graphstorm_processing/distributed_executor.py"
     S3_ENTRY_POINT="s3://${OUTPUT_BUCKET}/emr-scripts/distributed_executor.py"
+    aws s3 cp "${PATH_TO_ENTRYPOINT}" ${S3_ENTRY_POINT}
+
+    OUTPUT_PREFIX="s3://${OUTPUT_BUCKET}/gsprocessing/emr/${GRAPH_NAME}/"
 
-    export OUTPUT_PREFIX="s3://${OUTPUT_BUCKET}/gsprocessing/emr/${GRAPH_NAME}/"
-
-    spark-submit --master yarn \
-        --deploy-mode cluster \
-        --conf spark.executorEnv.YARN_CONTAINER_RUNTIME_TYPE=docker \
-        --conf spark.executorEnv.YARN_CONTAINER_RUNTIME_DOCKER_IMAGE=${IMAGE} \
-        --conf spark.executorEnv.PYSPARK_PYTHON="/.pyenv/shims/python" \
-        --conf spark.yarn.appMasterEnv.YARN_CONTAINER_RUNTIME_TYPE=docker \
-        --conf spark.yarn.appMasterEnv.YARN_CONTAINER_RUNTIME_DOCKER_IMAGE=${IMAGE} \
-        --conf spark.yarn.appMasterEnv.PYSPARK_PYTHON="/.pyenv/shims/python" \
-        --num-executors ${NUM_EXECUTORS} \
-        ${S3_ENTRY_POINT} \
-            --config-filename ${CONFIG_FILENAME} \
+    python "${SCRIPT_DIR}/submit_gsp_emr_step.py" \
+        --entry-point-s3 ${S3_ENTRY_POINT} \
+        --instance-type ${INSTANCE_TYPE} \
+        --log-uri "${OUTPUT_PREFIX}/spark-logs" \
```

Comment:
We may need to remove the `/` here. For my test case, I see the s3 subdirectory create another folder named `/`. It should be due to the redundant `/` here.

Reply:
OK removed it from the $OUTPUT_PREFIX